### ISSUE TITLE
h5read dimensions

### ISSUE THREAD

**Issue by rafaqz** [JuliaIO/HDF5.jl#576]
I'm wondering why `h5read(x, :key, (1:10, 5))` returns a 2 dimensional array.

Would it be more consistent with `view` and `getindex` to reduce the last dimension
and return a one dimensional array? `h5read(x, :key, (1:10, 5:5))` would
return a two dimensional array.

**Comment by rafaqz** [comment]
This also illustrates the issue clearly. I'm having to special case the first with `dropdims` to standardise the output.

```julia
julia> h5read(f, "A", (1:10,5,1))
10×1×1 Array{Float64,3}:
[:, :, 1] =
 0.6444758013805767  
 0.18046362703473418 
 0.03768195187818879 
 0.036838732026658105
 0.3749745812333558  
 0.2620131414223197  
 0.5865032567307125  
 0.09073663271286003 
 0.1682095239179957  
 0.1828346223001902  

julia> h5read(f, "A")[1:10,5,1]
10-element Array{Float64,1}:
 0.6444758013805767  
 0.18046362703473418 
 0.03768195187818879 
 0.036838732026658105
 0.3749745812333558  
 0.2620131414223197  
 0.5865032567307125  
 0.09073663271286003 
 0.1682095239179957  
 0.1828346223001902  
```

**Comment by musm** [comment]
Yeah we should fix this.

**Comment by musm** [comment]
One options is to just do

```
    dropdims(ret, dims =  tuple(findall(map(x -> length(x) == 1, indices))...))
```
in `_getindex`

any thoughts?

**Comment by kleinhenz** [comment]
closed by #603 unless I missed something.
